src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,58 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.FormatStyle;
+
+import duke.DukeException;
+import duke.Parser;
+
+/**
+ * Represents a Deadline Task that has a description and a due date.
+ */
+public class Deadline extends Task {
+    protected LocalDate duedate;
+
+    /**
+     * Constructor for Deadline task, loaded from the storage file.
+     * @param input Description of task including its due date.
+     * @param isDone Whether the task is marked or unmarked.
+     */
+    public Deadline(String input, boolean isDone) {
+        super(input, isDone);
+        this.symbol = 'D';
+        String[] temp = input.split(",");
+        this.description = temp[0];
+        LocalDate inputFormatter = LocalDate.parse(temp[1]);
+        this.duedate = inputFormatter;
+        this.duedateString = inputFormatter.format(DateTimeFormatter.ofLocalizedDate(FormatStyle.LONG));
```

Comment:
Seems like it's not following SLAP, but I personally don't think it's a big issue here.